### PR TITLE
Add a Makefile and fix byte-compilation warnings

### PR DESCRIPTION
**Pull request by leungbk**
----

#

### CODE REVIEW

File: dap-mode.el [hunk 1]
```diff
@@ -1240,7 +1241,14 @@ DEBUG-SESSION is the active debug session."
              #'equal))
 
 (defun dap--set-exception-breakpoints (debug-session callback)
-  (-let [(&dap-session 'initialize-result 'launch-args (&plist :type)) debug-session]
+  ;; XXX: for whatever reason, make byte-compile emits warnings when
```
Comment (kiennq) [inline dap-mode.el:1244]: This seems weird, the production code has been proven working, perhaps we need to tune/require/shuffle something for this to work?

**Comment by kiennq** [comment]
Do you want too update travis to use new make file or convert travis to Github CI?

File: dap-mode.el (hunk 1 above)
Comment (yyoncho) [inline dap-mode.el:1244]: most likely the definition of the destructoring function should be in eval-when-compile block

File: dap-mouse.el [hunk 2]
```diff
@@ -166,6 +167,8 @@ If there is an active selection - return it."
           (setq dap-mouse--hide-timer nil)
           (remove-hook 'post-command-hook #'dap-tooltip-post-tooltip)))))))
 
+(declare-function #'lsp-treemacs-render "ext:lsp-treemacs")
```
Comment (yyoncho) [inline dap-mouse.el:170]: lsp-treemacs is already a dependency, so no need from declare-function just require it.

**Review by yyoncho — APPROVED** [review]

**Comment by yyoncho** [comment]
Thank you!

**Pull request merged**